fix: use AbortSignal from TanStack Query in queryFn functions

harsh04044 opened the pull request:
## Proposed Changes

Fixes #14866 
- Replace `new AbortController().signal` with `signal` parameter from queryFn in all affected files
- Update queryFn signatures to accept `signal` parameter: `async ({ pageParam = 0, signal }) => {}`
- Update HistoricalRecordSelector interface to support signal parameter in custom queryFn implementations
- This enables proper request cancellation when components unmount or queries are invalidated, preventing unnecessary network traffic and potential memory leaks

Tagging: @ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate the bug or test the new feature. _(N/A - Code-level fix with no user-facing behavior changes)_
- [ ] Update [product documentation](https://docs.ohc.network). _(N/A - Internal code improvement, no API or feature changes)_
- [ ] Ensure that UI text is placed in I18n files. _(N/A - No UI text changes)_
- [ ] Prepare a screenshot or demo video for the changelog entry and attach it to the issue. _(N/A - No visual changes, this is a code-level improvement)_
- [ ] Request peer reviews.
- [ ] Complete QA on mobile devices. _(N/A - Code-level fix, no UI or behavior changes visible to users)_
- [ ] Complete QA on desktop devices. _(N/A - Code-level fix, no UI or behavior changes visible to users)_
- [ ] Add or update Playwright tests for related changes. _(Optional - Request cancellation behavior is difficult to test with Playwright, but existing tests should still pass)_

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Improvements**
  * Propagate cancellation signals into list and paginated queries for consistent request aborting and more responsive UI.
  * Standardized paginated response typing for observation lists to reduce inconsistencies.
  * Improved handling of missing measurement values for more reliable display.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

harsh04044 committed:
fix: use AbortSignal from TanStack Query in queryFn functions

Replace incorrect AbortController().signal usage with signal parameter
from TanStack Query's queryFn to enable proper request cancellation.

- Update useInfiniteQuery queryFn to accept and use signal parameter
- Update useQuery queryFn to accept and use signal parameter
- Update HistoricalRecordSelector interface to accept signal parameter
- Fix 10 files with incorrect AbortController usage in queryFn contexts

Fixes #14866

harsh04044 committed:
fix: remove duplicate PaginatedResponse interface

Use existing PaginatedResponse type from @/Utils/request/types instead of local duplicate

harsh04044 commented on src/components/Common/Charts/ObservationHistoryTable.tsx:
```diff
@@ -15,13 +15,19 @@ import {
+  next: string | null;
+  previous: string | null;
+  results: T[];
+}
```
i will fix  this in upcoming commit

harsh04044 commented on src/components/Patient/vitals/list.tsx:
```diff
@@ -9,9 +9,9 @@ import { EncounterAccordionLayout } from "@/components/Patient/EncounterAccordio
 import { PaginatedResponse } from "@/Utils/request/types";
 import { Code } from "@/types/base/code/code";
-import { ObservationListRead } from "@/types/emr/observation/observation";
+import { Observation, ObservationWithUser } from "@/types/emr/observation";
```
i will address it upcoming commit

harsh04044 replied:
check this i have solved this in my latest commit

harsh04044 commented on src/components/Patient/vitals/list.tsx:
```diff
@@ -62,10 +59,10 @@ function extractVitals(
       if (code.display) {
         const vitalField = ob.find(
-          (fields) => fields.main_code?.code === code.code,
+          (fields) => fields.main_code.code === code.code,
```
i will address this in upcoming commit

harsh04044 committed:
fix: add index.ts to export Observation and ObservationWithUser type aliases

Create index file in observation module to export type aliases for
Observation and ObservationWithUser, mapping them to ObservationListRead.
This fixes unresolved import errors in components using these types.

harsh04044 committed:
fix: add null-safety for optional main_code property

Use optional chaining when accessing main_code.code to prevent
runtime errors when main_code is null or undefined.

harsh04044 committed:
fix: use observationApi.list instead of patientApi.listObservations

Fix incorrect API usage introduced during rebase. Also add null-safety
for optional main_code property in ObservationHistoryTable.

harsh04044 committed:
fix: restore NoteManager.tsx and fix TypeScript errors

- Restore NoteManager.tsx to match upstream/develop (removed incorrect imports)
- Fix null-safety for main_code in observations and vitals
- Fix value property type in vitals/list.tsx to handle null values
- Remove value_quantity references (doesn't exist in type)
- Fix formatting issues

Fixes build errors introduced during rebase. All TypeScript errors related to AbortController changes are now resolved.

harsh04044 committed:
docs: add JSDoc comments to clarify Observation type aliases

- Add documentation explaining when to use Observation vs ObservationWithUser
- Clarifies semantic distinction between the two aliases
- Addresses CodeRabbit suggestion for better type documentation

harsh04044 commented on src/types/emr/observation/index.ts:
```diff
@@ -0,0 +1,16 @@
+ * Base observation type for general observation data.
+ * Use this type when user attribution (created_by, updated_by) is not the primary concern.
+ */
+export type Observation = ObservationListRead;
```
i will address this in my latest commit

harsh04044 committed:
docs: fix misleading JSDoc comment for Observation type

- Update JSDoc to clarify that all observations include user attribution fields
- Remove misleading statement about user attribution not being primary concern
- Note that ObservationListRead always contains created_by, updated_by, data_entered_by
- Addresses CodeRabbit suggestion about accurate type documentation

Jacobjeevan commented on src/types/emr/observation/index.ts:
why? Discard type changes.

harsh04044 replied:
Thanks for your review , i have correccted it in my latest commit

Jacobjeevan commented on src/pages/Encounters/tabs/observations.tsx:
```diff
@@ -156,16 +156,22 @@ export const EncounterObservationsTab = () => {
-                        {item.main_code?.display ||
-                          item.main_code?.code ||
-                          t("unknown")}
+                        {item.main_code?.display || item.main_code?.code}
```
let's keep unknown

harsh04044 replied:
okay

Jacobjeevan commented on src/components/Common/Charts/ObservationHistoryTable.tsx:
```diff
@@ -49,15 +49,15 @@ export const ObservationHistoryTable = ({
 
   const { data, isLoading, hasNextPage, fetchNextPage } = useInfiniteQuery<
-    PaginatedResponse<ObservationListRead>
+    PaginatedResponse<ObservationWithUser>
```
keep types as is

harsh04044 replied:
Thanks for your review , i have correccted it in my latest commit

harsh04044 committed:
refactor: address maintainer feedback on PR #14874

- Remove type aliases file (src/types/emr/observation/index.ts)
- Revert ObservationWithUser to ObservationListRead in all files
- Restore t('unknown') fallback in observations.tsx
- Update imports to use @/types/emr/observation/observation

Addresses maintainer feedback:
- Discard type changes (delete index.ts)
- Keep types as is (use ObservationListRead)
- Keep unknown fallback in observations display

harsh04044 committed:
refactor: remove redundant String() conversion in vitals list

- Remove String() wrapper from code.display key assignment
- Since code.display is already typed as string and checked for truthiness,
  the String() conversion is redundant
- Addresses CodeRabbit suggestion

nihal467 committed:
Merge branch 'develop' into fix/abort-controller-usage

nihal467 reviewed (changes requested):
<img width="1463" height="819" alt="Image" src="https://github.com/user-attachments/assets/4cd3e5ef-e0b2-422f-8a64-be47cff37b5c" />

- the values are shown twice in observation section once we submit the progress note questionnaire

harsh04044 commented:
> <img alt="Image" width="1463" height="819" src="https://private-user-images.githubusercontent.com/57055998/531315820-4cd3e5ef-e0b2-422f-8a64-be47cff37b5c.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjcyNzE4MDIsIm5iZiI6MTc2NzI3MTUwMiwicGF0aCI6Ii81NzA1NTk5OC81MzEzMTU4MjAtNGNkM2U1ZWYtZTBiMi00MjJmLThhNjQtYmU0N2NmZjM3YjVjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAxMDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMTAxVDEyNDUwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI5M2ZiNjYxZjVjNTRhMDAwMGMzNjc3MzI3MGU1OGU1Mzk4ZmVjZjA4ZWEyMjE3MDk1NmQxYWZhODVlOTg2OTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.cGWBjt9XZBdJpt3UKAHomllfCbRatLseMYLmLM-4IqY">
> * the values are shown twice in observation section once we submit the progress note questionnaire

i will address this in my upcoming commit

harsh04044 committed:
fix: remove duplicate observation value rendering
